packages/legend-query-builder/src/components/result/QueryBuilderResultPanel.tsx

```diff
@@ -269,6 +380,32 @@ export const QueryBuilderResultPanel = observer(
       resultState.isGeneratingPlan ||
       resultState.pressedRunQuery.isInProgress;
 
+    const resultLimit = Math.min(
+      queryBuilderState.fetchStructureState.implementation instanceof
+        QueryBuilderTDSState &&
+        queryBuilderState.fetchStructureState.implementation
+          .resultSetModifierState.limit
+        ? queryBuilderState.fetchStructureState.implementation
+            .resultSetModifierState.limit
+        : Number.MAX_SAFE_INTEGER,
+      resultState.previewLimit,
+    );
+
+    const truncateExecutionResultIfExceed = (
+      execResult: ExecutionResult,
+    ): ExecutionResult => {
+      if (
+        execResult instanceof TDSExecutionResult &&
+        execResult.result.rows.length > resultLimit
+      ) {
+        const truncatedExecutionResult = deepClone(execResult);
```

Comment:
is deepClone needed here ? on every render ?
why can't we save it in the state. Maybe have an ExecutionResultState with executionResult and other flags like has overflow?